packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/simulationservice/SimulationRequestResource.java

```diff
@@ -177,5 +192,42 @@ public JobResponse makeEnsembleCalibrateCiemssJob(
 		return res;
 	}
 
-
+	//Get modelConfigId
+	//Check if it has timeseries in its metadata
+	//If it does for each element convert it to type Intervention and add it to this.interventions
+	//Schema: http://json-schema.org/draft-07/schema#
+	public List<Intervention> getInterventionFromId(String modelConfigId){
+		List<Intervention> interventionList = new ArrayList<>();
+		try {
+			ObjectMapper mapper = new ObjectMapper();
+			ModelConfiguration modelConfig = modelConfigProxy.getModelConfiguration(modelConfigId);
+			JsonNode configuration =  mapper.convertValue(modelConfig.getConfiguration(), JsonNode.class);
+			if (configuration.findValue("timeseries") != null){
+				JsonNode timeseries = mapper.convertValue(configuration.findValue("timeseries"), JsonNode.class);
+				List<String> fieldNames = new ArrayList<>();
+				timeseries.fieldNames().forEachRemaining(key -> fieldNames.add(key));
+				for (int i = 0; i < fieldNames.size(); i++){
+					// Eg) Beta
+					String interventionName = fieldNames.get(i).replaceAll("\"",",");
+					// Eg) "1:0.14, 10:0.1, 20:0.2, 30:0.3"
+					String tempString = timeseries.findValue(fieldNames.get(i)).toString().replaceAll("\"","").replaceAll(" ","");
```

Comment:
what are we trying to do here? If you need to resort to regex here it usually means we are on the wrong path.

Reply:
Parse this junk which at some point needs a redesign
![image](https://github.com/DARPA-ASKEM/Terarium/assets/17088680/f2fc9381-47d9-4c18-85e0-a63005e05a05)